hal/src/stm32f2xx/sleep_hal.cpp

```diff
@@ -186,6 +302,42 @@ static int validateWakeupSource(hal_sleep_mode_t mode, const hal_wakeup_source_b
     return SYSTEM_ERROR_NOT_SUPPORTED;
 }
 
+static void enterSleepSleep() {
```

Comment:
This should really have a better name :sweat_smile: Doesn't really tell you what it does.

Do we want a separate function? As far as I can see the only difference is that we may not need to call `PWR_EnterSTOPMode()` and keep HSE/HSI running. 

Let's also add comments on why we prefer HSI to HSE here. I wouldn't expect a lot of difference in power consumption.

Reply:
From the comment: https://github.com/particle-iot/device-os/pull/2173/files?file-filters%5B%5D=#r476527209, which approach would you prefer?

Reply:
Honestly, we could probably go either way. If I had to choose, I'd probably keep HSE.